pkg/network/network.go

```diff
@@ -370,7 +378,7 @@ func KnativeProxyHeader(r *http.Request) string {
 // IsProbe returns true if the request is a Kubernetes probe or a Knative probe,
 // i.e. non-empty ProbeHeaderName header.
 func IsProbe(r *http.Request) bool {
-	return IsKubeletProbe(r) || KnativeProbeHeader(r) != ""
+	return IsKubeletProbe(r) || KnativeProbeHeader(r) != "" || isScrapeRequestOfPrometheus(r)
```

Comment:
Mostly looks good, but  my obviously confusing naming comment was that this is no longer `IsProbe` but rather `IsProbeOrMetric` or `IsOutOfBoundsRequest` (not sure what a good name would be).